### DISCUSSION TITLE
Understanding how to use Notifications

### DISCUSSION THREAD

**Discussion by MyGuyCai** [saleweaver/python-amazon-sp-api#538] [Q&A] [answered]
I'm trying to wrap my head around notifications.
What I want to do is let amazon tell me when there is a price change for any product that I currently have listed.

I followed the docs and have done these steps:
1. Created a destination
```
note = Notifications(credentials=credentials, marketplace=Marketplaces.UK)
note.create_destination(name='test2', arn=credentials['role_arn'])
```
2. Created the subscription
```
id = None
ds = note.get_destinations()
for destination in ds.payload:
    if destination['name'] == "test":
        id = destination['destinationId']
if id:
    note.create_subscription(NotificationType.ANY_OFFER_CHANGED, destination_id=id)
```
3. Tried to get the notification
```
res = note.get_subscription(NotificationType.ANY_OFFER_CHANGED)
print(res)
```

The payload I'm returned is:
```
{'destinationId': 'XX-XX-XX-XX-XX',
 'payloadVersion': '1.0',
 'processingDirective': None,
 'subscriptionId': 'XX-XX-XX-XX-XX'}
```

Is there something I'm missing about getting the actual price changes?

**Comment by saleweaver** [comment, marked as answer]
The destination has to be a SQS Queue, not a role. 

The notifications are explained here:

https://docs.developer.amazonservices.com/en_UK/subscriptions/Subscriptions_ReceivingNotifications.html

Once you've set this up you can listen to the queue and process messages that Amazon pushed to it.

**Reply by MyGuyCai** [reply]
Thanks for getting back to me, and sorry for the late reply.
I went ahead and deleted the destination and subscription that were using the wrong arn.
I headed over to [AWS SQS](https://eu-west-2.console.aws.amazon.com/sqs/v2/home?region=eu-west-2#/queues) and created an Amazon SQS queue with all default settings.
In python I created a destination and subscription using the newly created arn. 
I will wait to see if the queue works as expected now.

**Reply by saleweaver** [reply]
All default settings may be the error. You need to add permissions for Amazon to send to this queue

**Reply by MyGuyCai** [reply]
Yep, just found out how to format the acess policy json. I believe I got this right 
I've now added
```
{
      "Sid": "__sender_statement",
      "Effect": "Allow",
      "Principal": {
        "AWS": "arn:aws:iam::437568002678:root"
      },
      "Action": [
        "SQS:SendMessage",
        "SQS:GetQueueAttributes"
      ],
      "Resource": "arn:aws:sqs:region:account:quename"
    }
```
To my Access Policy

**Reply by saleweaver** [reply]
You have to adapt the string in the sender code. (Arn)
Have a look at the example, it's all in there.

**Reply by saleweaver** [reply]
Region, accountid, and so on

**Reply by MyGuyCai** [reply]
This seems to have fixed it, it's now displaying as 223 messages available.
Thanks for the help, great work you're doing here.

**Reply by MyGuyCai** [reply]
> Region, accountid, and so on

Yeah just formatted it like that to hide my details, thanks.
Do you have a 'buy me a coffee' link?

**Reply by saleweaver** [reply]
I don't. Always happy for a GitHub one time sponsoring though. 
Even if you don't, happy to have helped.

**Comment by saleweaver** [comment]
Here's a meeting should you need help. https://calendly.com/d/dkx-8jn-nzp/sp-api
